pkg/apply/applydrivers/apply_drivers_default.go

```diff
@@ -112,7 +113,7 @@ func (c *Applier) initStatus() {
 
 // todo: atomic updating status after each installation for better reconcile?
 // todo: set up signal handler
-func (c *Applier) updateStatus(err error) {
+func (c *Applier) updateStatus(errCode constants.ErrorCode, err error) {
```

Comment:
Return two error, one for app apply, another for cluster update is better.
As in golang, error can be any type that implmented   `Error()`, Not return custom type, but return error type with custom errors.
Like:
```golang
var ErrNotMatch = errors.New("NotMatch")

func do() error {
    return ErrNotMatch
}

if errors.Is(err, ErrNotMatch) {
    // RODO
}
```